ledger-tool/src/main.rs

```diff
@@ -798,21 +797,28 @@ fn load_bank_forks(
     };
 
     if let Some(halt_slot) = process_options.halt_at_slot {
-        for slot in starting_slot..=halt_slot {
-            if let Ok(Some(slot_meta)) = blockstore.meta(slot) {
-                if !slot_meta.is_full() {
-                    eprintln!("Unable to process from slot {} to {} due to blockstore slot {} not being full",
-                        starting_slot, halt_slot, slot);
-                    exit(1);
+        let mut max_slot = 0;
+        let mut next_slots: VecDeque<_> = vec![starting_slot].into();
+        while !next_slots.is_empty() {
+            let next_slot = next_slots.pop_front().unwrap();
+            if let Ok(Some(slot_meta)) = blockstore.meta(next_slot) {
+                if slot_meta.is_full() {
+                    max_slot = std::cmp::max(max_slot, next_slot);
+                    if max_slot >= halt_slot {
+                        break;
+                    }
+
+                    next_slots.extend(slot_meta.next_slots);
```

Comment:
Could this lead to a potential false positive on being able to process? For example, consider the following fork graph:
```
                 slot 2
                 /    \
            slot 3    |
               |      |
               |   slot 4
               |      |
              maj    min
```
4 could be on a minor (the wrong) fork, but still be a child of 2 (maybe the leader that produced 4 didn't get 3 in time and assumed that it was skipped). Suppose that we wanted to halt on 3 but didn't have the full slot; if we have a full slot 4, this check would say we're good to go.

I think we want to check for `== halt_slot`. However, we also want to quit when all possible candidates in `next_slots` are `> halt_slot`. I didn't check this for compile, but I think the following tweak should do the trick (please double check this):
```rust
        let mut can_process = false;
        while !next_slots.is_empty() {
            let next_slot = next_slots.pop_front().unwrap();
            if let Ok(Some(slot_meta)) = blockstore.meta(next_slot) {
                if slot_meta.is_full() {
                    if slot_meta.slot() == halt_slot {
                        can_process = true;
                        break;
                    } else if slot_meta.slot() < halt_slot {
                        next_slots.extend(slot_meta.next_slots);
                    } else {
                        // this slot is too big, so all children would be as well
                    }
                } 
            }
        }
```

Reply:
Yeah I think you're right that mine could have had a false positive in that case. This code seems correct, I'll add it in and test it